File: php/features.php
<?php
declare(strict_types=1);

// NhlApiDocumentation SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class NhlApiDocumentationFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new NhlApiDocumentationBaseFeature();
            case "test":
                return new NhlApiDocumentationTestFeature();
            default:
                return new NhlApiDocumentationBaseFeature();
        }
    }
}

File: py/features.py
# NhlApiDocumentation SDK feature factory

from feature.base_feature import NhlApiDocumentationBaseFeature
from feature.test_feature import NhlApiDocumentationTestFeature


def _make_feature(name):
    features = {
        "base": lambda: NhlApiDocumentationBaseFeature(),
        "test": lambda: NhlApiDocumentationTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
